org.eclipse.ice.dev/org.eclipse.ice.dev.dependencyscraper/src/main/java/org/org/eclipse/ice/dev/dependencyscraper/DependencyScraper.java

```diff
@@ -0,0 +1,227 @@
+	 */
+	private boolean shouldInclude(ZipEntry file) {
+		return includes.stream()
+			.anyMatch(
+				include -> FilenameUtils.wildcardMatch(file.getName(), include)
+			);
+	}
+
+	/**
+	 * Copy file from jar to output directory.
+	 * @param jar from which file will be copied
+	 * @param fileInJar file to copy
+	 * @throws MojoFailureException if file could not be copied, inadequate
+	 *         permissions, etc.
+	 */
+	private void copyFileInJarToOutput(
+		JarFile jar, ZipEntry fileInJar
+	) throws MojoFailureException {
+		String fullPath = fileInJar.getName();
+		// Drop sourceDirectory from file path
+		String relativePath = fullPath
+			.substring(fullPath.toLowerCase(Locale.ENGLISH)
+				.indexOf(sourceDirectory.toLowerCase(Locale.ENGLISH))
+				+ sourceDirectory.length());
+		// Determine output path
+		File target = new File(outputDirectory, relativePath);
+		try {
+			if (target.exists()) {
+				if (clobber) { // Overwrite existing file
+					getLog().info(String.format(
+						"File %s already exists and clobber is set; overwriting.",
+						target.getName()
+					));
+
+					// Replace target only if contents differ
+					File tempFile = File.createTempFile(fullPath, null);
+					FileUtils.copyInputStreamToFile(
+						jar.getInputStream(fileInJar), tempFile
+					);
+					if (!FileUtils.contentEquals(tempFile, target)) {
+						FileUtils.forceDelete(target);
+						FileUtils.moveFile(tempFile, target);
+					} else {
+						tempFile.delete();
+					}
+				} else {
+					getLog().info(String.format(
+						"File %s already exists and clobber is not set; skipping.",
+						target.getName()
+					));
+				}
+			} else {
+				FileUtils.copyInputStreamToFile(
+					jar.getInputStream(fileInJar),
+					target
+				);
+			}
+		} catch (IOException e) {
+			throw new MojoFailureException(
+				"Failed to copy file " + fileInJar.getName()
+			);
+		}
+	}
+
+	/**
+	 * Collect Jar files from dependencies and search through them for files
+	 * matching the parameters of the plugin. Copy matching files into output
+	 * directory specified by parameters.
+	 */
+	public void execute() throws MojoExecutionException, MojoFailureException {
+		if (jarFiles == null) {
+			this.jarFiles = project.getArtifacts().stream()
+				.filter(artifact -> "jar".equals(artifact.getType()))
+				.map(Artifact::getFile).collect(Collectors.toSet());
+		}
+		for (File jar : jarFiles) {
+			try (JarFile jarFile = new JarFile(jar, false)) {
+				Set<ZipEntry> toCopy = jarFile.stream()
+					.filter(file -> !file.isDirectory())
+					.filter(this::startsWithSourceDirectory)
+					.filter(this::shouldInclude)
+					.collect(Collectors.toSet());
+				for (ZipEntry file : toCopy) {
+					getLog().info(String.format("Copying %s", file.getName()));
+					copyFileInJarToOutput(jarFile, file);
+				}
+			} catch (IOException e) {
+				throw new MojoFailureException(
+					String.format("Failed to open jar file %s!", jar.toString())
+				);
+			}
+		}
```

Comment:
More documentation would be nice, even if only a few lines. All of the big functional blocks, like 205-207 and 211-214 should have at least a few lines describing what they are doing because functional code is inherently confusing. Personally I would put one line per filter in addition a one line initial description.
```java
// Get the set of files to copy
Set<ZipEntry> toCopy = jarFile.stream()
   // No directories
   .filter(file -> !file.isDirectory())
   // Is in the source directory
   .filter(this::startsWithSourceDirectory)
   // Matches the inclusion filter for this type
  .filter(this::shouldInclude).collect(Collectors.toSet());
```